app/static/js/base.js

```diff
@@ -5,45 +5,113 @@
             for (let resource in added_resource){
                 let resource_amount = $userResources.data('army-'+resource);
                 $(`#current-${resource}-amount`).text(resource_amount + added_resource[resource].amount);
                 $('#user-resources').data(`army-${resource}` ,resource_amount + added_resource[resource].amount);
             }
-            $btn.data('turns-amount', turns);
+            $userResources.data('army-turns', turns);
             $('#current-turns-amount').text(`turns: ${turns}`);
         }
 
+        function setSpinner($btn, {added_resource}) {
+            let $card_body = $btn.closest('.card').find('.card-body');
+            $card_body.empty();
+            if (added_resource['field'])
+                $card_body.append('<h5 class="card-title">Search for field</h5>');
+            else
+                $card_body.append('<h5 class="card-title">Search for resources</h5>');
+            $card_body.append(`<div class="spinner">
+                                <div class="rect1"></div>
+                                <div class="rect2"></div>
+                                <div class="rect3"></div>
+                                <div class="rect4"></div>
+                                <div class="rect5"></div>
+                            </div>`);
+            }
+
         function setCardBody($btn, {added_resource}) {
             let $card_body = $btn.closest('.card').find('.card-body');
             $card_body.empty();
-            $card_body.append('<h5 class="card-title">Search for resources</h5>');
+            if (added_resource['field'])
+                $card_body.append('<h5 class="card-title">Search for field</h5>');
+            else
+                $card_body.append('<h5 class="card-title">Search for resources</h5>');
             $card_body.append('<p class="card-text"> you found: </p><div class="d-flex"></div>');
             for (let resource in added_resource) {
-                $card_body.find('div').append('<div class="col text-center mt-2"></div>');
+                if (resource === 'diamond' && added_resource[resource].amount === 0)
+                    continue;
+                $card_body.find('div').append('<div class="col text-center mt-2 pl-0 pr-0"></div>');
                 $card_body.find('.col').last().append(`<p class="card-text mb-0"><img class="resource-img img-fluid" src="${added_resource[resource].picture}" /></p>`);
                 $card_body.find('.col').last().append(`<p class="card-text">${added_resource[resource].amount}</p>`);
             }
         }
 
+        function setPurchaseResultsMessage($purchaseResultsDiv, purchaseSuccess, text) {
+            $purchaseResultsDiv.removeClass('message-showing').empty();
+            $purchaseResultsDiv.append(`<p>${text}</p>`).addClass('message-showing');
+            if (purchaseSuccess)
+                $purchaseResultsDiv.find('p').addClass('bg-success');

```

Comment:
For identical lines with different values you can do this.
This checks if purchaseSuccess is true. If so, we use 'bg-success', else, we use 'bg-warning'
On google, search javascript ternary operator
```suggestion
                $purchaseResultsDiv.find('p').addClass(purchaseSuccess ? 'bg-success' : 'bg-warning');
```

Reply:
this is pretty nice, thanks